env0/resource_cost_credentials.go

```diff
@@ -0,0 +1,208 @@
+	case string(client.AzureCostCredentialsType):
+		payLoadToSent := payLoad.(client.AzureCredentialsCreatePayload)
+		payLoadToSent.Name = d.Get("name").(string)
+		apikey, err = apiClient.AzureCredentialsCreate(payLoadToSent)
+	case string(client.GoogleCostCredentiassType):
+		payLoadToSent := payLoad.(client.GoogleCostCredentialsCreatePayload)
+		payLoadToSent.Name = d.Get("name").(string)
+		apikey, err = apiClient.GoogleCostCredentialsCreate(payLoadToSent)
+	}
+
+	if err != nil {
+		return diag.Errorf("cost credential fialed: %v", err)
+	}
+	d.SetId(apikey.Id)
+	return nil
+}
+
+func resourceCostCredentialsRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	apiClient := meta.(client.ApiClientInterface)
+
+	id := d.Id()
+	_, err := apiClient.CloudCredentials(id)
+	if err != nil {
+		return diag.Errorf("could not get credentials: %v", err)
+	}
+	return nil
+
+}
+
+func resourceCostCredentialsDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	apiClient := meta.(client.ApiClientInterface)
+
+	id := d.Id()
+	err := apiClient.CloudCredentialsDelete(id)
+	if err != nil {
+		return diag.Errorf("could not delete credentials: %v", err)
+	}
+	return nil
+
+}
+
+func getCredType(d *schema.ResourceData) (string, error) {
+	_, awsOk := d.GetOk("arn")
+	_, azureOk := d.GetOk("client_id")
+	_, googleOk := d.GetOk("table_id")
+	switch {
+	case awsOk:
+		return string(client.AwsCostCredentialsType), nil
+	case azureOk:
+		return string(client.AzureCostCredentialsType), nil
+	case googleOk:
+		return string(client.GoogleCostCredentiassType), nil
+	default:
+		return "error", errors.New("error in schema, no required value  defined")
+	}
+
+}
+
+func setPayload(d *schema.ResourceData) (interface{}, string, error) {
+	credType, err := getCredType(d)
+	if err != nil {
+		return nil, "", err
+	}
+	switch credType {
+	case string(client.AwsCostCredentialsType):
+
+		return client.AwsCredentialsCreatePayload{
+			Type: client.AwsCostCredentialsType,
+			Value: client.AwsCredentialsValuePayload{
+				RoleArn:    d.Get("arn").(string),
+				ExternalId: d.Get("external_id").(string),
+			},
+		}, credType, nil
+	case string(client.AzureCostCredentialsType):
+		return client.AzureCredentialsCreatePayload{
+			Type: client.AzureCostCredentialsType,
+			Value: client.AzureCredentialsValuePayload{
+				ClientId:       d.Get("client_id").(string),
+				ClientSecret:   d.Get("client_secret").(string),
+				TenantId:       d.Get("tenant_id").(string),
+				SubscriptionId: d.Get("subscription_id").(string),
+			},
+		}, credType, nil
+	case string(client.GoogleCostCredentiassType):
+		return client.GoogleCostCredentialsCreatePayload{
+			Type: client.GoogleCostCredentiassType,
+			Value: client.GoogleCostCredentialsValeuPayload{
+				TableId: d.Get("table_id").(string),
+				Secret:  d.Get("secret").(string),
+			},
+		}, credType, nil
+
+	default:
+		return "", "", errors.New("cant create payload")
```

Comment:
```suggestion
		return "", "", errors.New("Failed to build credentials payload, please contact env0")
```

or something like that, since we won't know why they got here, if we didn't found the cred type we should fail before and terraform should protect us from missing attributes

Reply:
e2814aded3ee3fac6539413714d2b80bb490692e